src/us/kbase/workspace/database/Workspace.java

```diff
@@ -836,20 +837,15 @@ private void sortObjects(
 			reports.put(wo, rep);
 			idhandler.associateObject(new IDAssociation(objcount, true));
 			try {
-				for (final Provenance.ProvenanceAction action: wo.getProvenance().getActions()) {
+				for (final ProvenanceAction action: wo.getProvenance().getActions()) {
 					for (final String pref: action.getWorkspaceObjects()) {
-						if (pref == null) {
-							throw new TypedObjectValidationException(String.format(
-									"Object %s has a null provenance reference",
-									getObjectErrorId(wo, objcount)));
-						}
 						idhandler.addStringId(new IdReference<String>(WS_ID_TYPE, pref, null));
 					}
 				}
-			} catch (IdReferenceHandlerException ihre) {
-				throw new TypedObjectValidationException(String.format(
-						"Object %s has invalid provenance reference: ",
-						getObjectErrorId(wo, objcount)) + ihre.getMessage(), ihre);
+			} catch (IdReferenceHandlerException e) {
+				// thrown if ID is null or the empty string, which is impossible
+				throw new RuntimeException("This exception didn't actually happen. In fact "
+						+ "you're on extremely strong drugs. Pay me no heed", e);
```

Comment:
I've always wondered about the best way to handle impossible errors. Now I know! 😁

Reply:
An alternate approach: https://github.com/kbase/groups/blob/f9f17aacede77d71bf8d79ef98b5d653bc174163/src/us/kbase/groups/service/api/APICommon.java#L306-L307